control-plane/pkg/reconciler/broker/namespaced_broker.go

```diff
@@ -107,14 +119,65 @@ func (r *NamespacedReconciler) ReconcileKind(ctx context.Context, broker *eventi
 }
 
 func (r *NamespacedReconciler) FinalizeKind(ctx context.Context, broker *eventing.Broker) reconciler.Event {
+	namespaceLock := r.LockMap.GetLock(broker.Namespace)
+	namespaceLock.Lock()
+	defer namespaceLock.Unlock()
+
 	br := r.createReconcilerForBrokerInstance(broker)
 	result := br.FinalizeKind(ctx, broker)
 
 	r.IPsLister.Unregister(types.NamespacedName{Namespace: broker.Namespace, Name: broker.Name})
 
+	err := r.cleanUpClusterScopedResources(ctx, broker)
+	if err != nil {
+		return fmt.Errorf("failed to clean up cluster scoped resources for KafkaNamespaced brokers in the namespace: %w", err)
+	}
+
 	return result
 }
 
+// cleanUpClusterScopedResources deletes the cluster scoped resources created for the data plane in the namespace, if
+// this is the last namespaced broker in the namespace
+func (r *NamespacedReconciler) cleanUpClusterScopedResources(ctx context.Context, broker *eventing.Broker) error {
```

Comment:
should we reflect the comment on the name of the `func`? the name does not imply any speciality about being the last NS-'d broker

Reply:
I separated the code that checks the number of namespaced brokers and deletes the cluster scoped resources. Name of the function reflects it now.